File: src/TypeScript/Problem1.ts
function findMedianSortedArrays(nums1: number[], nums2: number[]): number {
    let finalArray: number[] = [];
    let i = 0, j = 0;
    while (i < nums1.length && j < nums2.length) {
        if (nums1[i] < nums2[j]) {
            finalArray.push(nums1[i]);
            i++
        }
        else {
            finalArray.push(nums2[j]);
            j++;
        }
    }
    while (i < nums1.length) {
        finalArray.push(nums1[i]);
        i++;
    }
    while (j < nums2.length) {
        finalArray.push(nums2[j]);
        j++;
    }
    const length = finalArray.length;
    const mid = Math.floor(length / 2);

    return length % 2 !== 0
    ? finalArray[mid]
    : (finalArray[mid - 1] + finalArray[mid]) /2;
}

File: src/Java/Problem1.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        List<Integer> finalArray = new ArrayList<>();
        int i = 0, j = 0;
        while(i < nums1.length && j < nums2.length) {
            if (nums1[i] < nums2[j]) {
                finalArray.add(nums1[i]);
                i++;
            }
            else {
                finalArray.add(nums2[j]);
                j++;
            }
    }
    while (i < nums1.length) {
        finalArray.add(nums1[i]);
        i++;
    }
    while (j < nums2.length) {
        finalArray.add(nums2[j]);
        j++;
    }
    int arrayLength = finalArray.size();
    if (arrayLength % 2 != 0) {
        return finalArray.get(arrayLength / 2);
    }
    else {
        return (finalArray.get(arrayLength / 2 - 1) + finalArray.get(arrayLength / 2)) / 2.0;
    }
}
}